### PR TITLE
[NVIDIA] Fix num_experts in modelopt_quant

### PR DESCRIPTION
**Pull request by wenscarl**
A hotfix for https://github.com/sgl-project/sglang/pull/8779. 
The trtllm_fp4_block_scale_moe API checks for the routing_logits dim to be consistent with num_experts(global). But previously at create_weight, the num_experts is overwritten by num_local_experts. 
cc. @kushanam  @zhyncs 

## Accuracy Test

With EP enabled:
Accuracy: 0.963
Invalid: 0.000
Latency: 697.093 s
Output throughput: 139.871 token/s

With TP enabled
Accuracy: 0.956
Invalid: 0.000
Latency: 636.827 s
Output throughput: 152.779 token/s
## Benchmark & Profiling

With shared expert fusion
Accuracy: 0.963
Invalid: 0.000
Latency: 633.068 s
Output throughput: 154.200 token/s

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/layers/quantization/modelopt_quant.py [hunk 1]
```diff
@@ -764,7 +765,7 @@ def create_weights(
             )
 
         # TODO(ch-wan): check if this is needed
-        layer.num_experts = num_experts
+        layer.num_experts = num_experts * get_moe_expert_parallel_world_size()
```
Comment (kaixih) [inline python/sglang/srt/layers/quantization/modelopt_quant.py]: Do we still need to differentiate num_experts (global) from local_num_experts?

e.g. in the following w13_weight allocation, we should use the local one and for the CutlassMoEParams, we should use num_experts (global)?

**Comment by kaixih** [comment]
It is a simple change to enable EP fp4 moe, which might affect the original TP case. So can we have both EP and TP accuracy numbers after this change? @wenscarl

File: python/sglang/srt/layers/quantization/modelopt_quant.py (hunk 1 above)
Comment (wenscarl) [reply python/sglang/srt/layers/quantization/modelopt_quant.py]: Yes. We still need differentiate them. The issue is at [call-site](https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/layers/moe/fused_moe_triton/layer.py#L202), the layer is passed in as `self`. And outside `create_weights` the global num_experts and num_local_experts have been set already.

**Comment by pavanimajety** [comment]
@wenscarl May I ask you please fix these unnecessary prints that were merged with #8552:
The model loading is very verbose

  -  https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/layers/quantization/modelopt_quant.py#L1109-L1129

**Comment by pavanimajety** [comment]
We can also perhaps test shared experts fusion with `--enable-trtllm-moe` and let the default behavior hold

**Review by ishandhanani — APPROVED** [review]
Tested with flashinfer 0.2.9 rc2. Lets remove those logs and get this in!

File: python/sglang/srt/layers/quantization/modelopt_quant.py (hunk 1 above)
Comment (kaixih) [reply python/sglang/srt/layers/quantization/modelopt_quant.py]: Talked offline: we should avoid updating layer members (num_experts/num_local_experts) inside this function.

**Review by pavanimajety — APPROVED** [review]
LGTM, thanks for making the changes!

**Review by ishandhanani — APPROVED** [review]

**Review by pavanimajety — APPROVED** [review]

**Review by kaixih — APPROVED** [review]
Noticed the accuracy results are already provided in the description. Approved. Thx.

File: python/sglang/srt/layers/quantization/modelopt_quant.py [hunk 2]
```diff
@@ -1106,7 +1104,7 @@ def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
                 layer.w13_weight_scale,
             )
 
-            print("Applied flashinfer weight processing for both w13 and w2")
+            logger.info_once("Applied flashinfer weight processing for both w13 and w2")
```
Comment (trevor-m) [inline python/sglang/srt/layers/quantization/modelopt_quant.py:1107]: Can we just remove these logs? Looks like debugging prints that got left in

Comment (pavanimajety) [reply python/sglang/srt/layers/quantization/modelopt_quant.py:1107]: Agree with removing or making it more meaningful, like `swapped w13 and w2` or `interleaving the weight tensors for flashinfer-trtllm-moe`

**Review by kushanam — APPROVED** [review]
